### PR TITLE
Conversation caterpillar: respect the has_avatar flag on authors

### PR DESCRIPTION
**Pull request by bluefuton**
In the Reader API endpoints, we add a boolean flag called `has_avatar`. We set this to false whenever the avatar is missing, or additionally if it's a generic avatar (the "mystery man").

This PR updates the conversation caterpillar to omit authors with `has_avatar` set to `false`.

### To test

Head to http://calypso.localhost:3000/devdocs/blocks/conversation-caterpillar and make sure Boris has been omitted from the avatars (sorry Boris). I've set `has_avatar: false` on his fixture but left the avatar URL intact.

### CODE REVIEW

**Comment by bluefuton** [comment]
@samouri Yep - for the moment, caterpillar only likes gravatars. 

What do you think about blavatar fallback @fraying?

**Comment by samouri** [comment]
my gut says no.  🚢 🚢

**Comment by blowery** [comment]
I'm not even sure what we'd fallback to. These are comment authors, not posts from a site where we have a fallback. Would it be the primary site for the author? Just seems complicated.

**Comment by blowery** [comment]
+1 to shipping it as it is

**Comment by fraying** [comment]
Definitely ship as is.

**Comment by bluefuton** [comment]
🚢

File: client/blocks/conversation-caterpillar/index.jsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ class ConversationCaterpillarComponent extends React.Component {
 		// Only display authors with a gravatar, and only display each author once
 		const uniqueAuthors = uniqBy( map( comments, 'author' ), 'ID' );
 		const displayedAuthors = takeRight(
-			filter( uniqueAuthors, 'avatar_URL' ),
+			filter( uniqueAuthors, 'has_avatar' ),
```
Comment (blowery) [inline client/blocks/conversation-caterpillar/index.jsx]: hmmm... should we filter on both?

Comment (bluefuton) [inline client/blocks/conversation-caterpillar/index.jsx]: I started with both, pondered, and then went back to `has_avatar` only. We should never have an avatar URL with `has_avatar` = `false` handed back from the API, should we?

**Pull request merged**